runtime/src/test/java/com/flipkart/flux/dao/TestWorkflow.java

```diff
@@ -50,6 +52,14 @@ public Integer dummyTask(String integer) {
                 e.printStackTrace();
             }
         }
+        if(shouldCancel) {
+            try {
+                throw new FluxCancelPathException();
+            } catch (Exception e){
+                System.out.println("Failed to cancel path.");
```

Comment:
This will be always be caught, check if the exception raised is not an instance of FluxCancelPath, something like this.
```
if( !e instanceof FluxCancelPathException){
 print ...
}
```

Reply:
Done.